### PR TITLE
For story 2100

### PR DESCRIPTION
**Pull request by yeunga**
Added rollback. Other changes for the story are in cadc-util (DBUtil createJNDIDataSource bug fix), cadc-vos-server and cadc-uws-server.

### CODE REVIEW

File: cadc-vosi/src/main/java/ca/nrc/cadc/vosi/avail/CheckDataSource.java [hunk 1]
```diff
@@ -141,13 +148,26 @@ public void check()
...
-                rs = st.executeQuery(testSQL);
-                rs.next(); // just check the result set, but don't care if there are any rows
+                if (testSQL.trim().toLowerCase().startsWith("select "))
+                {
+                    rs = st.executeQuery(testSQL);
+                    rs.next(); // just check the result set, but don't care if there are any rows
+                }
+                else
+                {
+                    int count = st.executeUpdate(testSQL);
```
Comment (brianmajor) [inline cadc-vosi/src/main/java/ca/nrc/cadc/vosi/avail/CheckDataSource.java]: I don't see why you have captured the update count here

File: cadc-vosi/src/main/java/ca/nrc/cadc/vosi/avail/CheckDataSource.java [hunk 2]
```diff
@@ -168,6 +188,20 @@ public void check()
         }
         finally
         {
+            if (this.rollback && con != null)
+            {
+                try
+                {
+                    con.rollback();
+                    con.setAutoCommit(true);
```
Comment (brianmajor) [inline cadc-vosi/src/main/java/ca/nrc/cadc/vosi/avail/CheckDataSource.java]: Why is this connection set to auto commit after the rollback happens?

File: cadc-vosi/src/main/java/ca/nrc/cadc/vosi/avail/CheckDataSource.java [hunk 3]
```diff
@@ -168,6 +188,20 @@ public void check()
...
+                }
+                catch (SQLException e)
+                {
+                    log.debug("rollback failed: " + dataSourceName + " (" + testSQL + ")", e);
+                    log.warn("rollback failed: " + dataSourceName + " (" + testSQL + ")");
```
Comment (brianmajor) [inline cadc-vosi/src/main/java/ca/nrc/cadc/vosi/avail/CheckDataSource.java]: There should only be one log message here... log.error with the stack trace.

File: cadc-vosi/src/main/java/ca/nrc/cadc/vosi/avail/CheckDataSource.java (hunk 1 above)
Comment (yeunga) [inline cadc-vosi/src/main/java/ca/nrc/cadc/vosi/avail/CheckDataSource.java]: No need to capture it. I put it in when I was testing the code. I can remove 'int count = '.

File: cadc-vosi/src/main/java/ca/nrc/cadc/vosi/avail/CheckDataSource.java (hunk 2 above)
Comment (yeunga) [inline cadc-vosi/src/main/java/ca/nrc/cadc/vosi/avail/CheckDataSource.java]: The autoCommit was set to false earlier. So to be complete, just revert it to its original state.

File: cadc-vosi/src/main/java/ca/nrc/cadc/vosi/avail/CheckDataSource.java (hunk 3 above)
Comment (yeunga) [inline cadc-vosi/src/main/java/ca/nrc/cadc/vosi/avail/CheckDataSource.java]: Sure. I can remove 'log.warn...'. I was just following the 'catch (SQLException e) {...} above. Maybe we should remove the 'log.warn ...' there as well.

**Pull request merged**